homeassistant/components/nmap_tracker/__init__.py

```diff
@@ -1 +1,372 @@
+            devices.ipv4_last_mac.pop(device.ipv4, None)
+        del devices.config_entry_owner[mac_address]
+
+
+def signal_device_update(mac_address) -> str:
+    """Signal specific per nmap tracker entry to signal updates in device."""
+    return f"{DOMAIN}-device-update-{mac_address}"
+
+
+class NmapDeviceScanner:
+    """This class scans for devices using nmap."""
+
+    def __init__(self, hass, entry, devices):
+        """Initialize the scanner."""
+        self.devices = devices
+        self.home_interval = None
+
+        self._hass = hass
+        self._entry = entry
+
+        self._scan_lock = None
+        self._stopping = False
+        self._scanner = None
+
+        self._entry_id = entry.entry_id
+        self._hosts = None
+        self._options = None
+        self._exclude = None
+
+        self._finished_first_scan = False
+        self._last_results = []
+        self._mac_vendor_lookup = None
+
+    async def async_setup(self):
+        """Set up the tracker."""
+        config = self._entry.options
+        self._hosts = cv.ensure_list_csv(config[CONF_HOSTS])
+        self._exclude = cv.ensure_list_csv(config[CONF_EXCLUDE])
+        self._options = config[CONF_OPTIONS]
+        self.home_interval = timedelta(
+            minutes=cv.positive_int(config[CONF_HOME_INTERVAL])
+        )
+        self._scan_lock = asyncio.Lock()
+        if self._hass.state == CoreState.running:
+            await self._async_start_scanner()
+            return
+
+        self._entry.async_on_unload(
+            self._hass.bus.async_listen(
+                EVENT_HOMEASSISTANT_STARTED, self._async_start_scanner
+            )
+        )
+
+    @property
+    def signal_device_new(self) -> str:
+        """Signal specific per nmap tracker entry to signal new device."""
+        return f"{DOMAIN}-device-new-{self._entry_id}"
+
+    @property
+    def signal_device_missing(self) -> str:
+        """Signal specific per nmap tracker entry to signal a missing device."""
+        return f"{DOMAIN}-device-missing-{self._entry_id}"
+
+    @callback
+    def _async_get_vendor(self, mac_address):
+        """Lookup the vendor."""
+        oui = self._mac_vendor_lookup.sanitise(mac_address)[:6]
+        return self._mac_vendor_lookup.prefixes.get(oui)
+
+    @callback
+    def _async_stop(self):
+        """Stop the scanner."""
+        self._stopping = True
+
+    async def _async_start_scanner(self, *_):
+        """Start the scanner."""
+        self._entry.async_on_unload(self._async_stop)
+        self._entry.async_on_unload(
+            async_track_time_interval(
+                self._hass,
+                self._async_scan_devices,
+                timedelta(seconds=TRACKER_SCAN_INTERVAL),
+            )
+        )
+        self._mac_vendor_lookup = AsyncMacLookup()
+        with contextlib.suppress((asyncio.TimeoutError, aiohttp.ClientError)):
+            # We don't care of this fails since its only
```

Comment:
We don't care if this fails since it only

Reply:
fixed in https://github.com/home-assistant/core/pull/54715